internal/abi/fastly/types.go

```diff
@@ -500,8 +501,177 @@ const (
 
 // witx:
 //
-//	(typename $object_store_handle (handle))
-type objectStoreHandle handle
+//	(typename $kv_store_handle (handle))
+//	(typename $kv_store_lookup_handle (handle))
+//	(typename $kv_store_insert_handle (handle))
+//	(typename $kv_store_delete_handle (handle))
+//	(typename $kv_store_list_handle (handle))
+type (
+	kvstoreHandle       handle
+	kvstoreLookupHandle handle
+	kvstoreInsertHandle handle
+	kvstoreDeleteHandle handle
+	kvstoreListHandle   handle
+)
+
+const (
+	invalidKVStoreHandle  = kvstoreHandle(math.MaxUint32 - 1)
```

Comment:
Sorry for not knowing why these are 1 less than the max as the sentinel value. Why is that?

Reply:
To match the Rust SDK
```
fastly-shared/src/lib.rs
23:pub const INVALID_KV_PENDING_LOOKUP_HANDLE: u32 = std::u32::MAX - 1;
24:pub const INVALID_KV_PENDING_INSERT_HANDLE: u32 = std::u32::MAX - 1;
25:pub const INVALID_KV_PENDING_DELETE_HANDLE: u32 = std::u32::MAX - 1;
26:pub const INVALID_KV_PENDING_LIST_HANDLE: u32 = std::u32::MAX - 1;
29:pub const INVALID_KV_STORE_HANDLE: u32 = std::u32::MAX - 1;
```